plugins/chat/spec/system/receiving_message_spec.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+RSpec.describe "Receiving message", type: :system, js: true do
+  fab!(:current_user) { Fabricate(:admin) }
+  fab!(:user_2) { Fabricate(:user) }
+  fab!(:user_3) { Fabricate(:user) }
+  fab!(:direct_message_channel_1) do
+    Fabricate(:direct_message_channel, users: [current_user, user_2])
+  end
+  fab!(:direct_message_channel_2) do
+    Fabricate(:direct_message_channel, users: [current_user, user_2, user_3])
+  end
+
+  let(:chat_page) { PageObjects::Pages::Chat.new }
+
+  before do
+    chat_system_bootstrap(current_user, [direct_message_channel_1, direct_message_channel_2])
+    sign_in(current_user)
+  end
+
+  context "when core sidebar is enabled" do
+    before do
+      SiteSetting.enable_sidebar = true
+      SiteSetting.enable_experimental_sidebar_hamburger = true
+    end
+
+    context "when receiving a direct message" do
+      it "moves the channel to the top" do
```

Comment:
```suggestion
      it "reorders the direct messages sidebar section links ordering by last message received" do
```

What do you think about this description instead? I feel like just saying move to the top is very vague because it does not provide the context of where "top" is.

Reply:
> I think it'll be helpful to write a more detailed commit message here. I couldn't tell this was for sidebar until I read the tests written for this bug.

It's not for sidebar... at least not for sidebar only. I added a test for sidebar only because it gives me enough confidence for now, and I want to move a lot of acceptance test to system spec. This sort method has been here almost forever but never had any test.

Reply:
I see. I think providing this context in the commit message or PR description will certainly be helpful.

Reply:
> What do you think about this description instead? I feel like just saying move to the top is very vague because it does not provide the context of where "top" is.

Im not super fan :D

1) not sure I want to rewrite "sidebar" when we are in a sidebar block
2) Im not sure we want to got into the details of the implementation here, which is also more complex than just last message received

Reply:
Although I do agree top is not super clear 🤔

Reply:
I think I will go with your change on this, I don't have much better to propose for now.

Reply:
I'm not sure I understand what you mean here with regards to my review that "moves the channel to the top" is very vague. Just based on the description, I can't tell where "top" is. Only after reading through the implementation detail of the test can I tell that "top" here refers to the sidebar section for direct messages.